core/pkg/service/flag-evaluation/flag_evaluator_v2.go

```diff
@@ -0,0 +1,274 @@
+package service
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	evalV1 "buf.build/gen/go/open-feature/flagd/protocolbuffers/go/flagd/evaluation/v1"
+	"connectrpc.com/connect"
+	"github.com/open-feature/flagd/core/pkg/evaluator"
+	"github.com/open-feature/flagd/core/pkg/logger"
+	"github.com/open-feature/flagd/core/pkg/service"
+	"github.com/open-feature/flagd/core/pkg/telemetry"
+	"github.com/rs/xid"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/trace"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+type FlagEvaluationService struct {
+	logger                *logger.Logger
+	eval                  evaluator.IEvaluator
+	metrics               *telemetry.MetricsRecorder
+	eventingConfiguration *eventingConfiguration
+	flagEvalTracer        trace.Tracer
+}
+
+// NewFlagEvaluationService creates a FlagEvaluationService with provided parameters
+func NewFlagEvaluationService(log *logger.Logger,
+	eval evaluator.IEvaluator,
+	eventingCfg *eventingConfiguration,
+	metricsRecorder *telemetry.MetricsRecorder,
+) *FlagEvaluationService {
+	return &FlagEvaluationService{
+		logger:                log,
+		eval:                  eval,
+		metrics:               metricsRecorder,
+		eventingConfiguration: eventingCfg,
+		flagEvalTracer:        otel.Tracer("flagd.evaluation.v1"),
+	}
+}
+
+// nolint:dupl,funlen
+func (s *FlagEvaluationService) ResolveAll(
+	ctx context.Context,
+	req *connect.Request[evalV1.ResolveAllRequest],
+) (*connect.Response[evalV1.ResolveAllResponse], error) {
+	reqID := xid.New().String()
+	defer s.logger.ClearFields(reqID)
+
+	sCtx, span := s.flagEvalTracer.Start(ctx, "resolveAll", trace.WithSpanKind(trace.SpanKindServer))
+	defer span.End()
+
+	res := &evalV1.ResolveAllResponse{
+		Flags: make(map[string]*evalV1.AnyFlag),
+	}
+
+	evalCtx := map[string]any{}
+	if e := req.Msg.GetContext(); e != nil {
+		evalCtx = e.AsMap()
+	}
+
+	values := s.eval.ResolveAllValues(sCtx, reqID, evalCtx)
+	span.SetAttributes(attribute.Int("feature_flag.count", len(values)))
+	for _, value := range values {
+		// register the impression and reason for each flag evaluated
+		s.metrics.RecordEvaluation(sCtx, value.Error, value.Reason, value.Variant, value.FlagKey)
+		switch v := value.Value.(type) {
+		case bool:
+			res.Flags[value.FlagKey] = &evalV1.AnyFlag{
```

Comment:
evalV1 in a v2 file reads weirdly but when we will remove the support for the old proto we can rename this file :)